internal/services/exchange.service.go

```diff
@@ -0,0 +1,62 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"momentum-go-server/internal/models"
+	"net/http"
+	"time"
+)
+
+var currentTime = time.Now()
+var yesterday = currentTime.AddDate(0, 0, -1)
+
+func getNBUData(date string) []models.NBU {
+	var response []models.NBU
+
+	apiURL := fmt.Sprintf("https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?date=%s&json", date)
+
+	resp, err := http.Get(apiURL)
+
+	if err != nil {
+		log.Println("Error creating HTTP request:", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Println("Error reading HTTP response body:", err)
```

Comment:
Better put here return []models.NBU{}, err

Reply:
Fixed it. API provider services now return default structures on errors.